amundsen_application/static/js/config/config-types.ts

```diff
@@ -17,6 +17,7 @@ export interface AppConfig {
   userIdLabel?: string /* Temporary configuration due to lacking string customization/translation support */;
   issueTracking: IssueTrackingConfig;
   logoPath: string | null;
+  numberFormat: string | null;
```

Comment:
`Intl.NumberFormat` takes more options than just a string and also has multiple parameters. Our configuration needs to match that in order to allow folks to use the full capabilities: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat

In addition, I would suggest making this configuration more flexible to support the various types of number formatting. You'll see in the documentation above the number formatting has different styles, like for showing `decimal` (default), `percent`, `currency`, and `unit` for different units of measurement. As a tool meant to show data, I imagine we will have a use case in the future. So perhaps we could make this configuration a dictionary that supports being able to configure all of the styles:

Use Case: Imagine we have a future use case with percentages, which will need different formatting options. 
```
// Used in code would be: Intl.NumberFormat('en-US', { ...some optional options... })
numberFormatting: {
   decimal: {
        locale: 'en-US',
        options: {...some optional options...}
    },
   percentage: {
        locale: 'en-US',
        options: {...some optional options...}
    }
}
```

What do you think of structuring the configuration as something like above? We can refine the exact configuration the main point here about creating something that will be able to be more easily extended in a backwards compatible manner to cover number formatting use cases that are feasible for us to encounter in the future.

Lastly, all new configurations should have doc added: https://github.com/amundsen-io/amundsenfrontendlibrary/blob/master/docs/application_config.md

Reply:
yes you are right we can modify configurations so that it provides flexibility for future use cases if we may want to show different units in future. Only reason I did not use full capabilities of `Intl.NumberFormat` is that in current scenario column statistics only includes absolute values, like mean , median, stddev, etc. Data in any column with column statistics will always have numbers only even if it is storing a percentage or currency. But I can add the flexibility to make it easily extendable.